R/plotHeatmap.R

```diff
@@ -154,22 +154,22 @@ plotHeatmap <- function(object, features, columns = NULL,
                 col_name <- paste0("unnamed", i)
             }
             column_variables[[col_name]] <- colour_fac
-            column_colorings[[col_name]] <- col_scale
+            if (is.null(column_annotation_colors[[col_name]])) {
+                column_annotation_colors[[col_name]] <- col_scale
+            }
         }
 
-        if (!missing(column_annotation_colors)) {
-            column_colorings <- column_annotation_colors
-        }
         # No need to subset for 'columns' or 'order_columns_by',
         # as pheatmap::pheatmap uses the rownames to handle this for us.
         column_variables <- do.call(data.frame,
             c(column_variables, list(row.names=colnames(object))))
+        column_annotation_colors <- column_annotation_colors[colour_columns_by]
     } else {
-        column_variables <- column_colorings <- NULL
+        column_variables <- NULL
```

Comment:
Does `pheatmap` break if `column_variables=NULL` but `column_annotation_colors!=NULL`? Might want to defensively set the latter to `NULL` just in case.

Reply:
Doesn't seem to but yeah better to be defensive there